pdfarranger/pageutils.py

```diff
@@ -1059,6 +1059,190 @@ def get_offset(self):
+        else:
+            # Continue searching until text is found or all has been searched
+            self.rectangles = []
+            cnt = 0
+            while len(self.rectangles) == 0 and cnt < len(self.model):
+                self.npage = (self.npage + step) % len(self.model)
+                self.rectangles = self.find_text(self.npage, text)
+                cnt += 1
+                self.entry.set_progress_fraction(cnt / max(1, len(self.model)))
+                while Gtk.events_pending():
+                    Gtk.main_iteration()
+                if not self.get_search_mode():
+                    break
+            self.nrect = max(0, -step * len(self.rectangles) - 1)
+
+        if len(self.rectangles) > 0:
+            rectangle = self.rectangles[self.nrect]
+            self.show_find_results(self.npage, [rectangle])
+        self.entry.set_progress_fraction(0)
+
+    def find_text(self, npage, text):
+        if len(self.model) == 0:
+            return []
+        npage = min(npage, len(self.model) - 1)
+        page = self.model[npage][0].duplicate(incl_thumbnail=False)
+        self.page = page.duplicate(incl_thumbnail=False)
+        if len(page.layerpages) == 0:
+            doc = self.pdfqueue[page.nfile - 1]
+            poppler_page = doc.get_page(page.npage - 1)
+        else:
+            # Page has layers -> for simplicity export and search in exported pdf.
+            # This is slower but easier than recalculating rectangle coordinates.
+            page.angle = 0
+            page.scale = 1
+            page.crop = Sides()
+            page.hide = Sides()
+            doc, buf = get_in_memory_poppler_doc([page], self.pdfqueue)
+            poppler_page = doc.get_page(0)
+        if text != "":
+            self.text_old = text
+        rectangles = poppler_page.find_text(text)
+        rectangles = self.apply_crop(rectangles, poppler_page.get_size(), self.page.crop)
+        return rectangles
+
+    def apply_crop(self, rectangles, page_size, crop):
+        for r in rectangles:
+            r.x1 -= page_size[0] * crop.left
+            r.x2 -= page_size[0] * crop.left
+            r.y1 -= page_size[1] * crop.bottom
+            r.y2 -= page_size[1] * crop.bottom
+        rectangles = self.visible_rectangles(rectangles, page_size, crop)
+        return rectangles
+
+    def visible_rectangles(self, rectangles, page_size, crop):
+        """Remove rectangles which are cropped away"""
+        to_remove = []
+        for num, r in enumerate(rectangles):
+            if (r.x1 > page_size[0] * (1 - crop.left - crop.right) or
+                r.x2 < 0 or
+                r.y1 > page_size[1] * (1 - crop.top - crop.bottom) or
+                r.y2 < 0
+                ):
+                to_remove.append(num)
+        for num in reversed(to_remove):
+            del rectangles[num]
+        return rectangles
+
+    def close(self, _widget):
+        self.clear_find_results(unselect_all=False)
+
+
```

Comment:
This file is getting quite long, how would you feel about splitting it into multiple files? (I haven't looked deeper into whether that is possible.)

Reply:
Yes it's long. But I am not sure how we should split it. I think that will be for another PR.